src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java

```diff
@@ -870,4 +870,36 @@ private boolean isValidUserDirection(
             }, notificationGenerators
         );
     }
+
+    @Bean
+    public PreSubmitCallbackHandler<AsylumCase> paymentPaidLegalRepNotificationHandler(
+        @Qualifier("paymentPaidNotificationGenerator") List<NotificationGenerator> notificationGenerators) {
+
+        return new NotificationHandler(
+            (callbackStage, callback) -> {
+                AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+                State currentState = callback.getCaseDetails().getState();
+
+                boolean isCorrectAppealType = asylumCase
+                    .read(APPEAL_TYPE, AppealType.class)
+                    .map(type -> type == PA).orElse(false);
+
+                boolean isCorrectAppealTypeAndState =
+                    isCorrectAppealType
+                    && (currentState != State.APPEAL_STARTED
+                        || currentState != State.APPEAL_SUBMITTED
+                        || currentState != State.PENDING_PAYMENT
+                    );
+
+                String paymentStatus = asylumCase
+                    .read(AsylumCaseDefinition.PAYMENT_STATUS, String.class).orElse("");
+
+                return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+                       && callback.getEvent() == Event.MAKE_PAYMENT
```

Comment:
I have got `Event.PAYMENT_APPEAL` in case-api

Reply:
Updated for correct event name.